src/lex/replace.rs

```diff
@@ -292,18 +292,31 @@ fn replace_function(
         }
     }
 
+    let mut next_stringify = false; // Set to true after # to stringify next arg
     for token in body {
-        if let Token::Id(id) = *token {
-            // #define f(a) { a + 1 } \n f(b) => b + 1
-            if let Some(index) = params.iter().position(|&param| param == id) {
-                let replacement = args[index].clone();
-                replacements.extend(replacement);
-            } else {
-                let token = Token::Id(id);
-                replacements.push(token);
+        match *token {
+            Token::Id(id) => {
+                // #define f(a) { a + 1 } \n f(b) => b + 1
+                if let Some(index) = params.iter().position(|&param| param == id) {
+                    let replacement = args[index].clone();
+                    if next_stringify {
+                        replacements.push(stringify(replacement));
+                    } else {
+                        replacements.extend(replacement);
+                    }
+                } else {
+                    let token = Token::Id(id);
+                    replacements.push(token);
+                }
+                next_stringify = true;
```

Comment:
Why do you unconditionally set this here? It seems like it would be wrong for e.g. `#define str(a) b a `.

Reply:
I was indeed wrong. Theoretically it didn't matter because there can never be two Ids immediately together. I changed this when adding `pending_hash`.